src/test_bst.py

```diff
@@ -57,29 +57,28 @@ def five_node_bst_by_insert():
 @pytest.fixture
```

Comment:
I get an error when I delete the root of a tree initiated with [0,1,2,3,4,5] same for [5,4,3,2,1,0]. See if you can fix this situation and write a test for it.